File: swift-bridge/Sources/CGEventsBridge/CGScrollPhase.swift
import CoreGraphics

@_cdecl("cgscroll_phase_raw_value")
public func cgscrollPhaseRawValue(index: UInt32) -> UInt32 {
    switch index {
    case 0: return CGScrollPhase.began.rawValue
    case 1: return CGScrollPhase.changed.rawValue
    case 2: return CGScrollPhase.ended.rawValue
    case 3: return CGScrollPhase.cancelled.rawValue
    case 4: return CGScrollPhase.mayBegin.rawValue
    default: return UInt32.max
    }
}

File: src/cg_scroll_phase.rs
//! `CGScrollPhase` values used with `kCGScrollWheelEventScrollPhase`.

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum CGScrollPhase {
    Began,
    Changed,
    Ended,
    Cancelled,
    MayBegin,
}

impl CGScrollPhase {
    #[must_use]
    pub const fn raw(self) -> u32 {
        match self {
            Self::Began => 1,
            Self::Changed => 2,
            Self::Ended => 4,
            Self::Cancelled => 8,
            Self::MayBegin => 128,
        }
    }

    #[must_use]
    pub const fn from_raw(raw: u32) -> Option<Self> {
        match raw {
            1 => Some(Self::Began),
            2 => Some(Self::Changed),
            4 => Some(Self::Ended),
            8 => Some(Self::Cancelled),
            128 => Some(Self::MayBegin),
            _ => None,
        }
    }
}
